### ISSUE TITLE
Fix S2589 FP: Condition actually isn't always false

### ISSUE THREAD

**Issue by psychotoad** [SonarSource/sonar-dotnet#8057]
### Description

SonarLint reports S2589 on a line which isn't always false.

### Repro steps

The following function will terminate. But SonarLint falsely claims that the condition `counter >= maxRetries` is always false.
 
```
public static void Example()
{
    const int maxRetries = 5;
    var counter = 0;
    while (true)
    {
        counter++;
        if (counter >= maxRetries)
        {
            return;
        }
    }
}
```

### Expected behavior

The analyzer should not raise an issue.

### Actual behavior

The Analyzer reports `Change this condition so that it does not always evaluate to 'False'.`

### Related information

* C#/VB.NET Plugins version: Sonarlint 7.3.0.77872
* Visual Studio version: 2022 (17.7.4)
* MSBuild / dotnet version: .Net 7
* SonarScanner for .NET version (if used)
* Operating System: Windows 10

**Comment by mary-georgiou-sonarsource** [comment]
Hello @psychotoad ,

Thank you for reporting this.
This is a known FP due to how the symbolic execution engine handles loops (it visits them twice).
This is unfortunately a very hard issue to fix.

I'm closing this issue as a duplicate of this https://github.com/SonarSource/sonar-dotnet/issues/7849.
